### PR TITLE
Add load balancer service settings for seeds

### LINKED ISSUE #2318: Allow configuration of load balancer service annotations per seed
**What would you like to be added**:
Let's enhance the `gardencorev1beta1.Seed` object and allow to configure a list of annotations that Gardener will add to every `Service` of type `LoadBalancer` in this seed (e.g., the kube-apiserver services, ingresses load balancers, VPN load balancers, etc.).

#2315 will introduce a new `.spec.settings` field that can be enhanced:

```yaml
apiVersion: core.gardener.cloud/v1beta1
kind: Seed
metadata:
  name: my-seed
spec:
  provider:
    type: <provider-name> # e.g., aws, azure, gcp, ...
    region: europe-1
  ...
  settings:
    ...
    loadBalancerServices:
      annotations:
        service.beta.kubernetes.io/aws-load-balancer-ssl-cert: arn:aws:acm:xx-xxxx-x:xxxxxxxxx:xxxxxxx/xxxxx-xxxx-xxxx-xxxx-xxxxxxxxx #replace this value
        service.beta.kubernetes.io/aws-load-balancer-backend-protocol: http
```

The seed and shoot controller will simply add the given annotations to all load balancer services.

**Why is this needed**:
Any existing Kubernetes cluster can be connected to a Gardener system by either providing the kubeconfig or by deploying the Gardenlet into it. However, based on the underlying infrastructure it is sometimes required to deviate from the default load balancer configuration, e.g. in order to make them accessible or to configure certain properties in a different way.

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area usability
/kind enhancement
/priority normal
/size m
/topology seed

**What this PR does / why we need it**:
This PR allows to configure load balancer service annotations for services created in the seed as part of the new `.spec.settings` field in `Seed` resources.

**Which issue(s) this PR fixes**:
Fixes #2318

**Special notes for your reviewer**:
cc @RaphaelVogel 

**Release note**:

```improvement operator
It is now possible to configure annotations for `Service`s of type `LoadBalancer` created in the seed clusters. These annotations will be injected into each of these `Service` resources. They can be configured in the `.spec.settings.loadBalancerServiceAnnotations` field in `Seed` resources. Please consult [this document](https://github.com/gardener/gardener/blob/master/docs/usage/seed_settings.md#load-balancer-services) for more information.
```

### CODE REVIEW

File: pkg/apis/core/v1alpha1/types_seed.go [hunk 1]
```diff
@@ -193,6 +197,14 @@ type SeedSettingScheduling struct {
 	Visible bool `json:"visible" protobuf:"bytes,1,opt,name=visible"`
 }
 
+// SeedSettingLoadBalancerServices controls certain settings for services of type load balancer that are created in the
+// seed.
+type SeedSettingLoadBalancerServices struct {
+	// Annotations is a map of annotations that will be injected/merged into every load balancer service object.
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty" protobuf:"bytes,1,opt,name=annotations"`
```
Comment (timuthy) [inline pkg/apis/core/v1alpha1/types_seed.go]:
```suggestion
	Annotations map[string]string `json:"annotations,omitempty" protobuf:"bytes,1,rep,name=annotations"`
```
To be consistent with https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L238.

File: pkg/apis/core/v1beta1/types_seed.go [hunk 2]
```diff
@@ -193,6 +197,14 @@ type SeedSettingScheduling struct {
 	Visible bool `json:"visible" protobuf:"bytes,1,opt,name=visible"`
 }
 
+// SeedSettingLoadBalancerServices controls certain settings for services of type load balancer that are created in the
+// seed.
+type SeedSettingLoadBalancerServices struct {
+	// Annotations is a map of annotations that will be injected/merged into every load balancer service object.
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty" protobuf:"bytes,1,opt,name=annotations"`
```
Comment (timuthy) [inline pkg/apis/core/v1beta1/types_seed.go]:
```suggestion
	Annotations map[string]string `json:"annotations,omitempty" protobuf:"bytes,1,rep,name=annotations"`
```

**Review by timuthy — CHANGES_REQUESTED** [review]
Just nits. Looks good otherwise.

File: pkg/apis/core/v1alpha1/types_seed.go (hunk 1 above)
Comment (rfranzke) [reply pkg/apis/core/v1alpha1/types_seed.go]: Thanks, I'm wondering why this stuff was not automatically generated correctly...

**Review by timuthy — APPROVED** [review]
/lgtm

**Pull request merged**